### PR TITLE
Rust: array sum

### PR DESCRIPTION
**Pull request by Fleshgrinder**
This Rust solution allows arbitrary input dimensions.

### CODE REVIEW

**Comment by mre** [comment]
Could we additionally keep the old solution as it shows how to use (nested) generics to solve the problem?
It might not be the applicable to all input sizes, but I like that it simply uses traits and type constraints to express how to sum up any type that is `Sum`.

**Comment by Fleshgrinder** [comment]
I don’t think that this is the right place for showcasing Rust features or using it to teach people Rust. It is expected from an applicant that she knows Rust, otherwise we would not ask her to solve problems in it.

It’s your repository so of course I can revive it but it does not solve the problem at hand.

File: problems/array-sum/array-sum.rs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    }
+}
+
+fn main() {
+    assert_eq!(
+        15,
+        array_sum(vec![
+            Int(1),
+            Int(2),
+            List(vec![Int(3), Int(4), List(vec![Int(5)]),]),
```
Comment (Fleshgrinder) [inline problems/array-sum/array-sum.rs]: This weird formatting is the result of `rustfmt`.

**Comment by mre** [comment]
You have a point. When it comes to correctness versus elegance, we should always choose the former. 👍

**Pull request merged**